inference-engine/src/transformations/src/transformations/op_conversions/convert_interpolate1_to_interpolate4.cpp

```diff
@@ -0,0 +1,73 @@
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ConvertInterpolate1ToInterpolate4, "ConvertInterpolate1ToInterpolate4", 0);
+
+ngraph::pass::ConvertInterpolate1ToInterpolate4::ConvertInterpolate1ToInterpolate4() {
+    auto interpolate1 = ngraph::pattern::wrap_type<ngraph::opset1::Interpolate>();
+
+    ngraph::matcher_pass_callback callback = [this](pattern::Matcher& m) {
+        auto interpolate1 = std::dynamic_pointer_cast<ngraph::opset1::Interpolate>(m.get_match_root());
+        if (!interpolate1)
+            return false;
+
+        auto data_node = interpolate1->input_value(0);
+        auto out_shape_node = std::dynamic_pointer_cast<ngraph::opset1::Constant>(interpolate1->input_value(1).get_node_shared_ptr());
+        auto interpolate_attrs = interpolate1->get_attrs();
+        auto input_shape = data_node.get_shape();
+
+        // attrs
+        auto mode_v4 = ngraph::op::v4::Interpolate::InterpolateMode();
+        if (interpolate_attrs.mode == "nearest") {
+            mode_v4 = ngraph::op::v4::Interpolate::InterpolateMode::nearest;
+        } else if (interpolate_attrs.mode == "cubic") {
+            mode_v4 = ngraph::op::v4::Interpolate::InterpolateMode::cubic;
+        } else if (interpolate_attrs.mode == "linear") {
+            if (input_shape.size() < 5) {
+                mode_v4 = ngraph::op::v4::Interpolate::InterpolateMode::linear_onnx;
+            } else if (input_shape.size() == 5) {
+                mode_v4 = ngraph::op::v4::Interpolate::InterpolateMode::linear;
+            } else {
+                return false;
+            }
+        } else {
+            return false;
+        }
+        auto nearest_mode_v4 = ngraph::op::v4::Interpolate::NearestMode::floor;
+        auto shape_calculation_mode_v4 = ngraph::op::v4::Interpolate::ShapeCalcMode::sizes;
+        auto coordinate_transformation_mode_v4 = interpolate_attrs.align_corners ? ngraph::op::v4::Interpolate::CoordinateTransformMode::align_corners :
+                                                ngraph::op::v4::Interpolate::CoordinateTransformMode::asymmetric;
+        auto interpolate4_attr = ngraph::op::v4::Interpolate::InterpolateAttrs(mode_v4, shape_calculation_mode_v4,
+            interpolate_attrs.pads_begin, interpolate_attrs.pads_end,
+            coordinate_transformation_mode_v4, nearest_mode_v4, interpolate_attrs.antialias, -0.75);
+
+        // input
+        auto axes = interpolate_attrs.axes.to_vector();
+        auto axes_const = ngraph::opset4::Constant(ngraph::element::Type_t::i64, {axes.size()}, axes);
+        auto axes_node = std::make_shared<ngraph::opset4::Constant>(axes_const);
+        auto default_scales = std::vector<float>(axes.size(), 1.f);
+        auto scales = ngraph::opset4::Constant(ngraph::element::Type_t::f32, {axes.size()}, default_scales);
```

Comment:
You need to set Constant precision based on interpolate data input precision.
```suggestion
        auto scales_node = ngraph::opset4::Constant::create(interpolate1->input(0).get_element_type(), {axes.size()}, {1.f});
```

Reply:
Scales should be f32, not same as input precision. And we actually do not use this scales anymore, just a placeholder for convert. interpolate1 always convert to interpolate4 with sizes mode. i.e. we use target_shape not scales to calculate.